### ISSUE TITLE
`bun repl` crashes on Mac

### ISSUE THREAD

**Issue by DaleSeo** [oven-sh/bun#12131]
### What version of Bun is running?

1.1.16

### What platform is your computer?

Darwin 23.5.0 arm64 arm

### What steps can reproduce the bug?

run `bun repl`

### What is the expected behavior?

it works

### What do you see instead?

```sh
bun repl
============================================================
Bun v1.1.16 (bf7b327f) macOS Silicon
Args: "bun" "bun-repl"
Elapsed: 2ms | User: 3ms | Sys: 2ms
RSS: 2.82MB | Peak: 2.82MB | Commit: 1.07GB | Faults: 57

panic(main thread): Bus error at address 0x10521DBE0
oh no: Bun has crashed. This indicates a bug in Bun, not your code.

To send a redacted crash report to Bun's team,
please file a GitHub issue using the link below:

 https://bun.report/1.1.16/MB1bf7b327AA__+h+lEunlB2pY__A4Cg+tnkF

[1]    40022 trace trap  bun repl
```

**Comment by DaleSeo** [comment]
Downgrading to v1.1.15 fixes the issue.

```sh
$ bun repl
Welcome to Bun v1.1.15
Type ".help" for more information.
[!] Please note that the REPL implementation is still experimental!
    Don't consider it to be representative of the stability or behavior of Bun overall.
```

**Comment by dylan-conway** [comment]
This was fixed in #12070 and will be in bun `v1.1.17` later today. You can also try out the fix with canary (`bun upgrade --canary`)

**Comment by Jarred-Sumner** [comment]
Duplicate of #12070
